c/src/ml-api-service-remote.c

```diff
@@ -451,41 +495,45 @@ _mlrs_edge_event_cb (nns_edge_event_h event_h, void *user_data)
  * @brief Create edge handle.
  */
 static int
-_mlrs_create_edge_handle (nns_edge_h * edge_h, edge_info_s * edge_info)
+_mlrs_create_edge_handle (_ml_remote_service_s * remote_s,
+    edge_info_s * edge_info)
 {
   int ret = 0;
-  ret = nns_edge_create_handle (edge_info->topic, edge_info->conn_type,
-      edge_info->node_type, edge_h);
+  nns_edge_h edge_h = NULL;
+
+  ret = nns_edge_create_handle (edge_info->id, edge_info->conn_type,
+      edge_info->node_type, &edge_h);
 
   if (NNS_EDGE_ERROR_NONE != ret) {
     _ml_error_report ("nns_edge_create_handle failed.");
     return ret;
   }
 
-  ret = nns_edge_set_event_callback (*edge_h, _mlrs_edge_event_cb, NULL);
+  ret = nns_edge_set_event_callback (edge_h, _mlrs_edge_event_cb, remote_s);
   if (NNS_EDGE_ERROR_NONE != ret) {
     _ml_error_report ("nns_edge_set_event_callback failed.");
-    nns_edge_release_handle (*edge_h);
+    nns_edge_release_handle (edge_h);
     return ret;
   }
 
-  _mlrs_set_edge_info (edge_info, *edge_h);
+  _mlrs_set_edge_info (edge_info, edge_h);
 
-  ret = nns_edge_start (*edge_h);
+  ret = nns_edge_start (edge_h);
   if (NNS_EDGE_ERROR_NONE != ret) {
     _ml_error_report ("nns_edge_start failed.");
-    nns_edge_release_handle (*edge_h);
+    nns_edge_release_handle (edge_h);
     return ret;
   }
 
   if (edge_info->node_type == NNS_EDGE_NODE_TYPE_SUB) {
-    ret = nns_edge_connect (*edge_h, edge_info->dest_host,
-        edge_info->dest_port);
+    ret = nns_edge_connect (edge_h, edge_info->dest_host, edge_info->dest_port);
```

Comment:
I am just wondering. It looks that ```edge_h``` already contains edge_info. Then, Why are those other parameters for the host and port necessary?

Reply:
Right, since we set edge info, we don't need dest host and port as you said. This is duplicate work.
However, we will not change this because the user may use the connection without the set info.